### PR TITLE
Master mdl 71574 54

### PR DESCRIPTION
**Pull request by safatshahin**
PR for issue #54 #81 #50

### CODE REVIEW

File: question/bank/editquestion/classes/editquestion_column.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * Get the URL for editing a question as a link.
+     *
+     * @param int $questionid the question id.
+     * @return \moodle_url the URL, HTML-escaped.
+     */
+    public function edit_question_moodle_url($questionid) {
+        return new \moodle_url($this->editquestionurl, ['id' => $questionid]);
```
Comment (guillogo) [inline question/bank/editquestion/classes/editquestion_column.php:93]: moodle_url is imported already, we can remove the `\` or the `use moodle_url;`

**Review by guillogo — COMMENTED** [review]
hi @safatshahin these new plugins are working as expected, just a couple of things in the comments and also, can you change the order of the menu? Now is:

image

but edit question should be first.

thank you.

**Comment by guillogo** [comment]
Also @safatshahin can you check all the `moodle_exception` the throws are missing: `throw new moodle_exception`

**Comment by safatshahin** [comment]
hi @guillogo 
made the changes and also changed the name for the editquestion plugin.

**Comment by guillogo** [comment]
hi @safatshahin this looks good and work as expected. Just 1 thing, as Tim comment in tracker https://tracker.moodle.org/browse/MDL-71574?focusedCommentId=856113&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-856113, we 
 should back to editquestion name.

Thank you.
**Guillermo**

**Comment by safatshahin** [comment]
hi @guillogo 
changed the name also added some more code which i found on some places, includes renderer, yui etc.
please have a look now.

**Comment by guillogo** [comment]
Hi @safatshahin this has been tested with the last changes after last review. Also, the js refactoring works good and as expected.

**Comment by safatshahin** [comment]
hi @guillogo 
added privacy api, and also rebased against the latest base class.
thanks

**Pull request merged**